scripts/dags_migrate.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+
+if [ $# -ne 5 ] ; then
+    echo "Usage: dags_migrate.sh project project_secret_dir project_user hdfs_superuser hadoop_home"
+    exit 1
+fi
+
+PROJECT=$1
+PROJECT_DAGS_DIR=/srv/hops/airflow/dags/$2
+PROJECT_USER=$3
+HDFS_SUPERUSER=$4
+HADOOP_HOME=$5
+
+if [ -d "$PROJECT_DAGS_DIR" ]; then
+  echo "Copying dags for project: $PROJECT"
+  for DAG in "$PROJECT_DAGS_DIR"/*; do
+    if [ -f $DAG ] ; then
+      if [[ "$DAG" == *.py ]]; then
+        echo "Found possible dag $DAG"
+        # to make it idempotent
+        if ! grep -q "access_control" "$DAG"; then
+            echo "Writing access dag access_control rule"
+            # Put the project to the dag `project` role with read and edit permissions
+            sed -i '/default_args = args,/a \ \ \ \ access_control = {\n \ \ \ \ \ \ \ "'"${PROJECT}"'": {"can_dag_read", "can_dag_edit"},\n \ \ \ \ },' $DAG
+        fi
+      fi
+    fi
+  done
+
+  # Copy all the file in the project dags to hopsfs
+  ${HADOOP_HOME}/bin/hdfs dfs -copyFromLocal -f ${PROJECT_DAGS_DIR}/* /Projects/${PROJECT}/Airflow
```

Comment:
Generally we export the environment variable `HADOOP_USER_NAME` before running expat so even though we are running expat as `root` user, in HopsFS it will be `hdfs` and the certificates will work and all that.

That said I would suggest trying executing the hdfs commands as hdfs user as they might create log files belonging to root which will later cause issues. So try `su "$HDFS_SUPERUSER" -c "${HADOOP_HOME}/bin/hdfs dfs -copyFromLocal ..."`

You already have the `HDFS_SUPERUSER` variable which is not used anywhere

Reply:
It was like that before. See https://github.com/logicalclocks/expat/pull/109/commits/8a906e489f7bcfe0972465029b2fad5ba8c402ea#r1302221924. @SirOibaf

Reply:
Fabio's comment is on `-copyFromLocal -f`. My comment is on the user who is executing the command.

Reply:
I mean this one https://github.com/logicalclocks/expat/pull/109#discussion_r1302221924

Reply:
Oh sorry I missed that. Yes we do export the environment variable but I think it's safer to execute it as `hdfs` user regardless. Also, in the cloud we don't - which I will fix

Reply:
Correction. We do export in a wrapper script here https://github.com/logicalclocks/cloud-chef/blob/master/templates/default/run_expat.sh.erb#L8